### PR TITLE
PMM-6746 psmdb clusters

### PR DESCRIPTION
**Pull request by BupycHuk**
[PMM-6746](https://jira.percona.com/browse/PMM-6746)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1198

- [ ] Links to other linked pull requests (optional).
https://github.com/percona/pmm/pull/602
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: services/management/dbaas/psmdb_cluster_service_test.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+	"github.com/percona/pmm-managed/utils/tests"
+)
+
+const kubeconfTest = `
+	{
+		"apiVersion": "v1",
+		"kind": "Config",
+		"users": [
+			{
+				"name": "percona-xtradb-cluster-operator",
```
Comment (percona-csalguero) [inline services/management/dbaas/psmdb_cluster_service_test.go:45]: `xtradb` -> `PSMDB` here and below

**Review by percona-csalguero — CHANGES_REQUESTED** [review]

File: services/management/dbaas/psmdb_cluster_service_test.go [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+	})
+
+	//nolint:dupl
+	t.Run("BasicCreatePSMDBClusters", func(t *testing.T) {
+		s := NewPSMDBClusterService(db, dbaasClient)
+		mockReq := controllerv1beta1.CreatePSMDBClusterRequest{
+			KubeAuth: &controllerv1beta1.KubeAuth{
+				Kubeconfig: kubeconfTest,
+			},
+			Name: "third.pxc.test.percona.com",
```
Comment (askomorokhov) [inline services/management/dbaas/psmdb_cluster_service_test.go:143]: Validation should not allow names with dots: 

https://github.com/percona/pmm/blob/PMM-2.0/api/managementpb/dbaas/xtradb_cluster.proto#L156-L164

Comment (askomorokhov) [reply services/management/dbaas/psmdb_cluster_service_test.go:143]: same in other places.

**Review by askomorokhov — CHANGES_REQUESTED** [review]

**Review by percona-csalguero — APPROVED** [review]

File: services/management/dbaas/xtradb_cluster_service_test.go [hunk 3]
```diff
@@ -193,30 +192,14 @@ func TestXtraDBClusterService(t *testing.T) {
...
 	//nolint:dupl
 	t.Run("BasicUpdateXtraDBCluster", func(t *testing.T) {
 		s := NewXtraDBClusterService(db, dbaasClient)
 		mockReq := controllerv1beta1.UpdateXtraDBClusterRequest{
 			KubeAuth: &controllerv1beta1.KubeAuth{
-				Kubeconfig: kubeconfTest,
+				Kubeconfig: pxcKubeconfigTest,
 			},
-			Name: "third-pxc-test",
+			Name: "third.pxc.test.percona.com",
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply services/management/dbaas/xtradb_cluster_service_test.go]: fixed

File: services/management/dbaas/psmdb_cluster_service_test.go (hunk 2 above)
Comment (BupycHuk) [reply services/management/dbaas/psmdb_cluster_service_test.go:143]: oh, missed

File: services/management/dbaas/psmdb_cluster_service.go [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+	// 1. Get psmdb host and status
+	//  - Ex.: kubectl get -o=json PerconaServerMongoDB/<cluster_name>
+	// 2. Get root password:
+	//   - Ex.: kubectl get secret my-cluster-name-secrets -o json  | jq -r ".data.MONGODB_USER_ADMIN_PASSWORD" | base64 -d
+	resp := dbaasv1beta1.GetPSMDBClusterResponse{
+		ConnectionCredentials: &dbaasv1beta1.GetPSMDBClusterResponse_PSMDBCredentials{
+			Username: "userAdmin",
+			Password: "userAdmin123456",
+			Host:     fmt.Sprintf("%s-rs0.default.svc.cluster.local", req.Name),
+			Port:     27017,
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply services/management/dbaas/psmdb_cluster_service.go:103]: Do we really need to return it? We also don't return db name, btw

[the remark this answers is not in the document]
Comment (askomorokhov) [reply services/management/dbaas/psmdb_cluster_service.go:103]: Without it, we connect not to a cluster, but to only one instance of MongoDB.
About DB, name - it is by convention admin. But replica set can have any name.

[the remark this answers is not in the document]
Comment (BupycHuk) [reply services/management/dbaas/psmdb_cluster_service.go]: Added

**Review by askomorokhov — APPROVED** [review]

**Pull request merged**